security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -85,7 +85,11 @@ func (c *CitadelClient) CSRSign(csrPEM []byte, certValidTTLInSec int64) ([]strin
 	if err := c.reconnectIfNeeded(); err != nil {
 		return nil, err
 	}
-	ctx := metadata.NewOutgoingContext(context.Background(), metadata.Pairs("ClusterID", c.opts.ClusterID))
+	metaKV := []string{"ClusterID", c.opts.ClusterID}
+	if c.opts.CertSigner != "" {
+		metaKV = append(metaKV, security.CertSigner, c.opts.CertSigner)
```

Comment:
Should we set it in IstioCertificateRequest.Metadata? I am not sure we want to overload `metadata` so much (which is really HTTP headers) to encode information about the request. The current usage of cluster is a bit of a hack

Reply:
Thanks for the suggestion! added certSigner into IstioCertificateRequest.Metadata now

Reply:
moved to  IstioCertificateRequest.Metadata

Reply:
It is still set here as well, can you clean this up?

Reply:
sure

Reply:
cleaned up